### ISSUE TITLE
Scan Job failed unexpectedly after running 24 hours

### ISSUE THREAD

**Issue by RabeeaEgbareia** [nexB/scancode.io#1127]
I created a new project with a new scan with scan_codebase pipeline.
The scan took A LOT of time (24 hours !) and it crashed with a dead worker.

version:
![image](https://github.com/nexB/scancode.io/assets/128488364/8fc8776a-520d-4394-8b72-d0668d10c4ac)

Running the scancode.io using Docker in Linux - debian

The project contains scala project code with its 3rd parties modules (jars)

Last logs from worker container:
![image](https://github.com/nexB/scancode.io/assets/128488364/2cff5e74-d93b-4f80-bbd3-2f6fc26ea39f)

**Comment by pombredanne** [comment]
yes, if you have a very large codebase there could be cases where this takes a long time. 
A couple things:
- what are the spec of the machine? ... RAM CPU?
- there could be ways to speed up some scanning with matching, using a PurlDB instance
